components/brave_rewards/resources/page/components/app.tsx

```diff
@@ -43,12 +49,24 @@ export function App () {
     }
   }, [])
 
+  // On Android a native modal is displayed when this page is accessed and
+  // the user has not opted-in to Rewards. For backward-compatibility with
+  // the previous Android-specific settings page, display the page content
+  // underneath the modal. Note that this behavior will need to change when
+  // Android is updated to force the user through onboarding before
+  // displaying content.
+  const showOnboarding = () => {
+    return rewardsData.showOnboarding && !isAndroid
+  }
+
   return (
     <LayoutManager>
       <style.root id='rewardsPage'>
         <div ref={renderToolbar}></div>
         <AppErrorBoundary>
-          {!rewardsData.initializing && <Settings />}
+          {showOnboarding()
```

Comment:
I still curious if there's a way to do this without changing the component hierarchy. If we change the "data loading" `useEffect` callback in `settings.tsx` to re-run when `rewardsData.initialized` changes, I'm thinking that should reload the page state correctly.

Reply:
Reverted in 858f77a6e495026eb68a74ac62b0fcbc30c39b1a and added `useEffect` dependencies in 56f6e0be8476547141e96e198f944468c26ddab9